File: src/main/java/chapter5/schedulers/NewThreadSchedulerExample.java
package chapter5.schedulers;

import common.CommonUtils;
import common.Log;
import io.reactivex.Observable;
import io.reactivex.schedulers.Schedulers;


public class NewThreadSchedulerExample {

    public void basic() {
        String[] objs = {"1", "3", "5"};
        Observable.fromArray(objs)
                .doOnNext(data -> Log.it("Original data : " + data))
                .map(data -> "<<" + data + ">>")
                .subscribeOn(Schedulers.newThread())
                .subscribe(Log::it);
        CommonUtils.sleep(500);

        Observable.fromArray(objs)
                .doOnNext(data -> Log.it("Original data : " + data))
                .map(data -> "##" + data + "##")
                .subscribeOn(Schedulers.newThread())
                .subscribe(Log::it);
        CommonUtils.sleep(500);
    }

    public static void main(String[] args) {
        NewThreadSchedulerExample demo = new NewThreadSchedulerExample();
        demo.basic();
    }

}

File: src/main/kotlin/chapter5/schedulers/NewThreadSchedulerExample_.kt
package chapter5.schedulers

import common.CommonUtils_
import common.Log_
import io.reactivex.Observable
import io.reactivex.schedulers.Schedulers


class NewThreadSchedulerExample_ {

    fun basic() {
        val objs = arrayOf("1", "3", "5")
        Observable.fromArray(*objs)
                .doOnNext { Log_.it("Original data : $it") }
                .map { "<<$it>>" }
                .subscribeOn(Schedulers.newThread())
                .subscribe{ Log_.it(it) }
        CommonUtils_.sleep(500)

        Observable.fromArray(*objs)
                .doOnNext { Log_.it("Original data : $it") }
                .map { "##$it##" }
                .subscribeOn(Schedulers.newThread())
                .subscribe{ Log_.it(it) }
        CommonUtils_.sleep(500)
    }

}


fun main(args: Array<String>) {
    val demo = NewThreadSchedulerExample_()
    demo.basic()
}
